# Some goals we would like to achieve:

  • Find out if the price varies depending on location?

  • Identify whether a property's price changes with a change in a host's total listings?

  • Which type of property is most popular?

  • Are listings with higher number of amenities priced higher?

  • In this exploratory data analysis we could perform hypothesis testing, linear as well as Lasso regression, clustering using the K-Means algorithm.

  • We could analyze the results of various regression models, we will evaluate the importance of features and visualize the results.

In [2]:
#import ibraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline 

In [6]:
#load datasets
df_2019 = pd.read_csv('janeiro2019.csv', encoding='latin-1')
df_2020 = pd.read_csv('janeiro2020.csv', encoding='latin-1')

C:\Users\LENOVO\AppData\Local\Temp\ipykernel_16304\2891402771.py:2: DtypeWarning: Columns (61,62) have mixed types. Specify dtype option on import or set low_memory=False.
  df_2019 = pd.read_csv('janeiro2019.csv', encoding='latin-1')
C:\Users\LENOVO\AppData\Local\Temp\ipykernel_16304\2891402771.py:3: DtypeWarning: Columns (61,62,94) have mixed types. Specify dtype option on import or set low_memory=False.
  df_2020 = pd.read_csv('janeiro2020.csv', encoding='latin-1')


In [8]:
#view datasets
display(df_2019.head())
display(df_2020.head())

id                         listing_url     scrape_id last_scraped  \
0  17878  https://www.airbnb.com/rooms/17878  2.019010e+13   18/01/2019   
1  24480  https://www.airbnb.com/rooms/24480  2.019010e+13   18/01/2019   
2  25026  https://www.airbnb.com/rooms/25026  2.019010e+13   18/01/2019   
3  31560  https://www.airbnb.com/rooms/31560  2.019010e+13   18/01/2019   
4  34608  https://www.airbnb.com/rooms/34608  2.019010e+13   18/01/2019   

                                        name  \
0          Very Nice 2Br - Copacabana - WiFi   
1           Nice and cozy near Ipanema Beach   
2  Beautiful Modern Decorated Studio in Copa   
3           NICE & COZY 1BDR - IPANEMA BEACH   
4       São-Conrado House with Ocean View     

                                             summary  \
0  Please note that special rates apply for New Y...   
1  My studio is located in the best  of Ipanema. ...   
2  Our apartment is a little gem, everyone loves ...   
3  This nice and clean 1 bedroom apartment is loc...   
4                                                NaN   

                                               space  \
0  - large balcony which looks out on pedestrian ...   
1  The studio is located at Vinicius de Moraes St...   
2  This newly renovated studio (last renovations ...   
3  This nice and clean 1 bedroom apartment is loc...   
4     Its a House inside a very good condo with 2...   

                                         description experiences_offered  \
0  Please note that special rates apply for New Y...                none   
1  My studio is located in the best  of Ipanema. ...                none   
2  Our apartment is a little gem, everyone loves ...                none   
3  This nice and clean 1 bedroom apartment is loc...                none   
4     Its a House inside a very good condo with 2...                none   

                               neighborhood_overview  ... instant_bookable  \
0  This is the best spot in Rio. Everything happe...  ...                t   
1  The beach, the lagoon, Ipanema is a great loca...  ...                f   
2  Copacabana is a lively neighborhood and the ap...  ...                f   
3  Die Nachbarschaft von Ipanema ist super lebend...  ...                t   
4                                                NaN  ...                f   

  is_business_travel_ready          cancellation_policy  \
0                        f  strict_14_with_grace_period   
1                        f  strict_14_with_grace_period   
2                        f  strict_14_with_grace_period   
3                        f  strict_14_with_grace_period   
4                        f  strict_14_with_grace_period   

  require_guest_profile_picture require_guest_phone_verification  \
0                             f                                f   
1                             f                                f   
2                             t                                t   
3                             f                                f   
4                             f                                f   

   calculated_host_listings_count  \
0                               1   
1                               1   
2                               3   
3                               1   
4                               8   

   calculated_host_listings_count_entire_homes  \
0                                            1   
1                                            1   
2                                            3   
3                                            1   
4                                            5   

  calculated_host_listings_count_private_rooms  \
0                                            0   
1                                            0   
2                                            0   
3                                            0   
4                                            3   

   calculated_host_listings_count_shared_rooms  reviews_per_month  
0        

id                         listing_url       scrape_id last_scraped  \
0  17878  https://www.airbnb.com/rooms/17878  20200121213543   2020-01-22   
1  21280  https://www.airbnb.com/rooms/21280  20200121213543   2020-01-22   
2  25026  https://www.airbnb.com/rooms/25026  20200121213543   2020-01-22   
3  31560  https://www.airbnb.com/rooms/31560  20200121213543   2020-01-22   
4  35636  https://www.airbnb.com/rooms/35636  20200121213543   2020-01-22   

                                        name  \
0          Very Nice 2Br - Copacabana - WiFi   
1           Renovated Modern Apt. Near Beach   
2  Beautiful Modern Decorated Studio in Copa   
3           NICE & COZY 1BDR - IPANEMA BEACH   
4           Cosy flat close to Ipanema beach   

                                             summary  \
0  Pls note that special rates apply for Carnival...   
1  Immaculately renovated top-floor apartment ove...   
2  Our apartment is a little gem, everyone loves ...   
3  This nice and clean 1 bedroom apartment is loc...   
4  This cosy apartment  is just a few steps away ...   

                                               space  \
0  - large balcony which looks out on pedestrian ...   
1  Immaculately renovated top-floor apartment in ...   
2  This newly renovated studio (last renovations ...   
3  This nice and clean 1 bedroom apartment is loc...   
4  The location is extremely convenient, safe and...   

                                         description experiences_offered  \
0  Pls note that special rates apply for Carnival...                none   
1  Immaculately renovated top-floor apartment ove...                none   
2  Our apartment is a little gem, everyone loves ...                none   
3  This nice and clean 1 bedroom apartment is loc...                none   
4  This cosy apartment  is just a few steps away ...                none   

                               neighborhood_overview  ... instant_bookable  \
0  This is the best spot in Rio. Everything happe...  ...                t   
1  This is the best neighborhood in Zona Sul.  Fo...  ...                f   
2  Copacabana is a lively neighborhood and the ap...  ...                f   
3  Die Nachbarschaft von Ipanema ist super lebend...  ...                t   
4  The  apartment street is very quiet and safe ....  ...                f   

  is_business_travel_ready          cancellation_policy  \
0                        f  strict_14_with_grace_period   
1                        f  strict_14_with_grace_period   
2                        f  strict_14_with_grace_period   
3                        f  strict_14_with_grace_period   
4                        f  strict_14_with_grace_period   

  require_guest_profile_picture require_guest_phone_verification  \
0                             f                                f   
1                             f                                f   
2                             t                                t   
3                             f                                f   
4                             t                                t   

   calculated_host_listings_count  \
0                               1   
1                               1   
2                               3   
3                               1   
4                               1   

   calculated_host_listings_count_entire_homes  \
0                                            1   
1                                            1   
2                                            3   
3                                            1   
4                                            1   

  calculated_host_listings_count_private_rooms  \
0                                            0   
1                                            0   
2                                            0   
3                                            0   
4                                            0   

   calculated_host_listings_count_shared_rooms  reviews_per_month

In [9]:
print(df_2019.columns)
print(df_2020.columns)

Index(['id', 'listing_url', 'scrape_id', 'last_scraped', 'name', 'summary',
       'space', 'description', 'experiences_offered', 'neighborhood_overview',
       ...
       'instant_bookable', 'is_business_travel_ready', 'cancellation_policy',
       'require_guest_profile_picture', 'require_guest_phone_verification',
       'calculated_host_listings_count',
       'calculated_host_listings_count_entire_homes',
       'calculated_host_listings_count_private_rooms',
       'calculated_host_listings_count_shared_rooms', 'reviews_per_month'],
      dtype='object', length=106)
Index(['id', 'listing_url', 'scrape_id', 'last_scraped', 'name', 'summary',
       'space', 'description', 'experiences_offered', 'neighborhood_overview',
       ...
       'instant_bookable', 'is_business_travel_ready', 'cancellation_policy',
       'require_guest_profile_picture', 'require_guest_phone_verification',
       'calculated_host_listings_count',
       'calculated_host_listings_count_entire_homes',
     

In [13]:
new_df_2019 = df_2019[['name', 'neighborhood_overview', 'host_id', 'host_name', 'neighbourhood', 'property_type',
                      'room_type', 'accommodates', 'bathrooms', 'bedrooms', 'beds', 'bed_type', 'amenities','square_feet',
                       'price', 'security_deposit', 'review_scores_rating']
                     ]

In [14]:
new_df_2020 = df_2020[['name', 'neighborhood_overview', 'host_id', 'host_name', 'neighbourhood', 'property_type',
                      'room_type', 'accommodates', 'bathrooms', 'bedrooms', 'beds', 'bed_type', 'amenities','square_feet',
                       'price', 'security_deposit', 'review_scores_rating']
                     ]

In [15]:
new_df_2019

name  \
0                    Very Nice 2Br - Copacabana - WiFi   
1                     Nice and cozy near Ipanema Beach   
2            Beautiful Modern Decorated Studio in Copa   
3                     NICE & COZY 1BDR - IPANEMA BEACH   
4                 São-Conrado House with Ocean View     
...                                                ...   
35345  Apartamento para temp na vista mais linda do RJ   
35346  Gávea - Vista deslumbrante com todo o conforto   
35347                     Apartamento próximo a praia   
35348             Paraíso no Recreio dos Bandeirantes   
35349                   Suíte confortável em Ipanema   

                                   neighborhood_overview    host_id  \
0      This is the best spot in Rio. Everything happe...      68997   
1      The beach, the lagoon, Ipanema is a great loca...      99249   
2      Copacabana is a lively neighborhood and the ap...     102840   
3      Die Nachbarschaft von Ipanema ist super lebend...     135635   
4                                                    NaN     149407   
...                                                  ...        ...   
35345                                                NaN  237289465   
35346                                                NaN   97027425   
35347                                                NaN   58839922   
35348                                                NaN  131837955   
35349                                                NaN  227493197   

             host_name             neighbourhood property_type  \
0             Matthias                Copacabana   Condominium   
1                 Goya                   Ipanema     Apartment   
2              Viviane                Copacabana     Apartment   
3               Renata                   Ipanema     Apartment   
4              Ricardo                       NaN         House   
...                ...                       ...           ...   
35345          Cecilia                  Botafogo     Apartment   
35346  Carlos  Augusto                    Gávea     Apartment   
35347           Andrea                Copacabana     Apartment   
35348          Daniela  Recreio dos Bandeirantes     Apartment   
35349            Alena                   Ipanema     Apartment   

             room_type  accommodates  bathrooms  bedrooms  beds  bed_type  \
0      Entire home/apt             5        1.0       2.0   2.0  Real Bed   
1      Entire home/apt             2        1.0       1.0   1.0  Real Bed   
2      Entire home/apt             3        1.0       1.0   2.0  Real Bed   
3      Entire home/apt             3        1.0       1.0   2.0  Real Bed   
4      Entire home/apt            15        4.0       5.0   5.0  Real Bed   
...                ...           ...        ...       ...   ...       ...   
35345  Entire home/apt             5        2.5       3.0   4.0  Real Bed   
35346     Private room             2        1.0       1.0   1.0  Real Bed   
35347  Entire home/apt             5        2.0       1.0   1.0  Real Bed   
35348     Private room             1        1.0       1.0   1.0  Real Bed   
35349     Private room             2        1.0       1.0   2.0  Real Bed   

                                               amenities  square_feet  \
0      {TV,"Cable TV",Internet,Wifi,"Air conditioning...          NaN   
1      {TV,"Cable TV",Wifi,"Air conditioning","First ...          NaN   
2      {TV,"Cable TV",Internet,Wifi,"Air conditioning...          NaN   
3      {TV,"Cable TV",Internet,Wifi,"Air conditioning...          NaN   
4      {TV,"Cable TV",Internet,Wifi,"Air conditioning...          NaN   
...                                                  ...          ...   
35345  {TV,Wifi,Kitchen,"Smoking allowed",Gym,Elevato...          NaN   
35346  {TV,Wifi,"Air conditioning",Pool,Kitchen,"Free...          NaN   
35347  {TV,Wifi,Kitchen,Elevator,Washer,"Fire extingu...          NaN   
35348  {Pool,Kitchen,"Free parking on premises",Gym,

In [16]:
new_df_2019['neighbourhood'].value_counts(normalize = True)

Copacabana                  0.257324
Barra da Tijuca             0.182690
Ipanema                     0.091019
Recreio dos Bandeirantes    0.059490
Botafogo                    0.050118
                              ...   
Colégio                    0.000031
Parque Colúmbia            0.000031
Cavalcante                  0.000031
Mangueira                   0.000031
Engenheiro Leal             0.000031
Name: neighbourhood, Length: 100, dtype: float64

In [18]:
new_df_2019['neighbourhood']

0                      Copacabana
1                         Ipanema
2                      Copacabana
3                         Ipanema
4                             NaN
                   ...           
35345                    Botafogo
35346                      Gávea
35347                  Copacabana
35348    Recreio dos Bandeirantes
35349                     Ipanema
Name: neighbourhood, Length: 35350, dtype: object

In [20]:
new_df_2019[['neighbourhood','price']]

neighbourhood      price
0                    Copacabana    $296.00
1                       Ipanema    $225.00
2                    Copacabana    $161.00
3                       Ipanema    $243.00
4                           NaN  $3,498.00
...                         ...        ...
35345                  Botafogo    $599.00
35346                    Gávea    $251.00
35347                Copacabana    $195.00
35348  Recreio dos Bandeirantes    $101.00
35349                   Ipanema    $199.00

[35350 rows x 2 columns]

In [22]:
new_df_2019[['property_type','price']].value_counts()

property_type  price    
Apartment      $199.00      1187
               $300.00      1164
               $150.00      1093
               $251.00       983
               $498.00       897
                            ... 
Earth house    $1,873.00       1
               $1,251.00       1
               $1,000.00       1
Cottage        $652.00         1
Yurt           $169.00         1
Length: 2343, dtype: int64

In [24]:
new_df_2019['property_type'].value_counts()

Apartment                 27340
House                      3590
Condominium                1452
Serviced apartment          878
Loft                        657
Bed and breakfast           266
Hostel                      230
Guest suite                 180
Other                       138
Guesthouse                  119
Villa                       109
Townhouse                    67
Aparthotel                   55
Boutique hotel               47
Hotel                        37
Earth house                  26
Casa particular (Cuba)       23
Boat                         22
Cottage                      20
Tiny house                   18
Chalet                       17
Cabin                        12
Bungalow                     10
Nature lodge                  9
Resort                        4
Farm stay                     4
Treehouse                     3
Castle                        3
Campsite                      3
Tent                          2
Barn                          2
Camper/R

majority property_types are:
Apartment                 27340

House                      3590

Condominium                1452

Serviced apartment          878

Loft                        657

Bed and breakfast           266

Hostel                      230

Guest suite                 180

Other                       138

Guesthouse                  119

Villa                       109

Townhouse                    67

Aparthotel                   55

Boutique hotel               47

Hotel                        37

Earth house                  26

Casa particular (Cuba)       23

Boat                         22

Cottage                      20

Tiny house                   18

Chalet                       17


Cabin                        12

Bungalow                     10

Nature lodge                  9

Resort                        4

Farm stay                     4

Treehouse                     3

Castle                        3

Campsite                      3

Tent                          2

Barn                          2

In [40]:
new_df_2019[['property_type', 'amenities', 'price']]

property_type                                          amenities  \
0       Condominium  {TV,"Cable TV",Internet,Wifi,"Air conditioning...   
1         Apartment  {TV,"Cable TV",Wifi,"Air conditioning","First ...   
2         Apartment  {TV,"Cable TV",Internet,Wifi,"Air conditioning...   
3         Apartment  {TV,"Cable TV",Internet,Wifi,"Air conditioning...   
4             House  {TV,"Cable TV",Internet,Wifi,"Air conditioning...   
...             ...                                                ...   
35345     Apartment  {TV,Wifi,Kitchen,"Smoking allowed",Gym,Elevato...   
35346     Apartment  {TV,Wifi,"Air conditioning",Pool,Kitchen,"Free...   
35347     Apartment  {TV,Wifi,Kitchen,Elevator,Washer,"Fire extingu...   
35348     Apartment  {Pool,Kitchen,"Free parking on premises",Gym,E...   
35349     Apartment  {TV,Wifi,"Air conditioning",Kitchen,"Free park...   

           price  
0        $296.00  
1        $225.00  
2        $161.00  
3        $243.00  
4      $3,498.00  
...          ...  
35345    $599.00  
35346    $251.00  
35347    $195.00  
35348    $101.00  
35349    $199.00  

[35350 rows x 3 columns]

In [36]:
import seaborn as sns

C:\Users\LENOVO\anaconda3\lib\site-packages\seaborn\_decorators.py:36: FutureWarning: Pass the following variable as a keyword arg: x. From version 0.12, the only valid positional argument will be `data`, and passing other arguments without an explicit keyword will result in an error or misinterpretation.
  warnings.warn(


TypeError: Horizontal orientation requires numeric `x` variable.